app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  # , counter_cache: :posts_counter
+
+  has_many :comments
+  has_many :likes
+
+  scope :last_five_comments, lambda { |post_id|
+    Comment.where(post_id:).order('comments.created_at DESC').limit(5)
+  }
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- **[OPTIONAL]** It would be advisable to add an `after_save` active record callback for this method. This callback ensure that this method is called whenever a post is saved, so that the author's post counter is updated without having to call this method directly.

  ```
  after_save :update_posts_counter
  ...

  private

  def update_posts_counter
    author.update(posts_counter: author.posts.count)
  end
  ```